### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by xiv**
This is a PR to turn on Github's code analysis feature. Findings will show under the security tab above.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
+      if: ${{ github.event_name == 'pull_request' }}
+      
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      # Override language selection by uncommenting this and choosing your languages
+      # with:
+      #   languages: go, javascript, csharp, python, cpp, java
```
Comment (cheshireoctopus) [inline .github/workflows/codeql-analysis.yml:32]: @xiv - do we have to specify `javascript` here?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
```
Comment (cheshireoctopus) [inline .github/workflows/codeql-analysis.yml:45]: Should this be `following two lines`?

**Review by cheshireoctopus — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (xiv) [reply .github/workflows/codeql-analysis.yml:32]: it should auto-detected without a manual specification

Comment (cheshireoctopus) [reply .github/workflows/codeql-analysis.yml:32]: Cool 👍

**Pull request merged**